### PR TITLE
Add a test for YAML 1.1 types

### PR DESCRIPTION
**Pull request by perlpunk**
This is the first step for supporting the YAML 1.2 Core schema.

https://github.com/perlpunk/yaml-test-schema has lots of test data for all the various YAML schemas/versions.

This PR adds
* the testdata for YAML 1.1 to the test/data directory
* a test loading and dumping all test data and check the results

Known bugs like missing boolean resolution for `y`, `n` etc. are skipped.

~~Also the test data file from https://github.com/perlpunk/yaml-test-schema is not yet in master.~~

### CODE REVIEW

**Comment by perlpunk** [comment]
@ingydotnet could you please have a look?
This is just a draft, but if it's principally ok, I can move forward with the next steps for 1.2.
Considered that you are still interested in 1.2 support.

I added an issue with suggestions https://github.com/yaml/pyyaml/issues/486

**Comment by perlpunk** [comment]
This PR is now also used for #497

**Review by Thom1729 — APPROVED** [review]

File: tests/lib/test_schema.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    if (int(expected) == value):
+        return 1
+    print(value)
+    print(expected)
+    return 0
+
+def check_float(value, expected):
+    if expected == 'inf()':
+        if value == float("inf"):
+          return 1
```
Comment (ingydotnet) [inline tests/lib/test_schema.py:24]: indent 4?

**Review by ingydotnet — CHANGES_REQUESTED** [review]
LGTM. Just fix indent to 4.

File: tests/lib/test_schema.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+def test_implicit_resolver(data_filename, skip_filename, verbose=False):
+    types = {
+        'str':   [str,   check_str],
+        'int':   [int,   check_int],
+        'float': [float, check_float],
+        'inf':   [float, check_float],
+        'nan':   [float, check_float],
+        'bool':  [bool,  check_bool],
+    }
+    skipdata = yaml.load(open(skip_filename, 'rb'), Loader=yaml.SafeLoader)
```
Comment (Thom1729) [inline tests/lib/test_schema.py:61]:
```suggestion
    with open(skip_filename, 'rb') as file:
        skipdata = yaml.load(file, Loader=yaml.SafeLoader)
```

File: tests/lib/test_schema.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    skip_load = skipdata['load'];
+    skip_dump = skipdata['dump'];
+    if verbose:
+        print(skip_load)
+    tests = yaml.load(open(data_filename, 'rb'), Loader=yaml.SafeLoader)
```
Comment (Thom1729) [inline tests/lib/test_schema.py:66]:
```suggestion
    with open(data_filename, 'rb') as file:
        tests = yaml.load(file, Loader=yaml.SafeLoader)
```

File: tests/lib/test_schema.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+    i = 0;
+    fail = 0;
+    for input, test in sorted(tests.items()):
+        if verbose:
+            print('-------------------- ' + str(i))
+        test = tests[input]
```
Comment (Thom1729) [inline tests/lib/test_schema.py:73]: Isn't this a no-op?

File: tests/lib/test_schema.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
+import yaml
+import sys
+import pprint
+
+def check_bool(value, expected):
+    if expected == 'false()' and value is False:
+        return 1
```
Comment (Thom1729) [inline tests/lib/test_schema.py:7]: Any reason to return 1/0 instead of true/false?

File: tests/lib/test_schema.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
+import yaml
+import sys
+import pprint
+
```
Comment (Thom1729) [inline tests/lib/test_schema.py:4]:
```suggestion
import math

```

File: tests/lib/test_schema.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+def check_int(value, expected):
+    if (int(expected) == value):
+        return 1
+    print(value)
+    print(expected)
+    return 0
+
+def check_float(value, expected):
+    if expected == 'inf()':
+        if value == float("inf"):
```
Comment (Thom1729) [inline tests/lib/test_schema.py:23]:
```suggestion
        if value == math.inf:
```

File: tests/lib/test_schema.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+          return 1
+    elif expected == 'inf-neg()':
+        if value == float("-inf"):
```
Comment (Thom1729) [inline tests/lib/test_schema.py:26]:
```suggestion
        if value == -math.inf:
```

File: tests/lib/test_schema.py [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+          return 1
+    elif expected == 'nan()':
+        if value != value or (value == 0.0 and value == 1.0):
```
Comment (Thom1729) [inline tests/lib/test_schema.py:29]:
```suggestion
        if math.isnan(value):
```

File: tests/lib/test_schema.py [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+    skipdata = yaml.load(open(skip_filename, 'rb'), Loader=yaml.SafeLoader)
+    skip_load = skipdata['load'];
+    skip_dump = skipdata['dump'];
+    if verbose:
+        print(skip_load)
+    tests = yaml.load(open(data_filename, 'rb'), Loader=yaml.SafeLoader)
+
+    i = 0;
+    fail = 0;
+    for input, test in sorted(tests.items()):
```
Comment (Thom1729) [inline tests/lib/test_schema.py:70]: If `i` is still needed:

```suggestion
    for i, (input, test) in enumerate(sorted(tests.items())):
```

File: tests/lib/test_schema.py [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+        if verbose:
+            print('-------------------- ' + str(i))
+        test = tests[input]
+        i += 1
+
+        # Skip known loader bugs
+        if input in skip_load:
+            continue
+
+        exp_type = test[0];
```
Comment (Thom1729) [inline tests/lib/test_schema.py:80]: Semicolons not necessary to terminate statements.

File: tests/lib/test_schema.py (hunk 1 above)
Comment (perlpunk) [reply tests/lib/test_schema.py]: done

**Review by Thom1729 — APPROVED** [review]

File: tests/lib/test_schema.py (hunk 6 above)
Comment (perlpunk) [reply tests/lib/test_schema.py:5]: done

File: tests/lib/test_schema.py (hunk 2 above)
Comment (perlpunk) [reply tests/lib/test_schema.py]: done

File: tests/lib/test_schema.py (hunk 3 above)
Comment (perlpunk) [reply tests/lib/test_schema.py]: done

File: tests/lib/test_schema.py (hunk 4 above)
Comment (perlpunk) [reply tests/lib/test_schema.py]: Yeah, probably a leftover, done

File: tests/lib/test_schema.py (hunk 10 above)
Comment (perlpunk) [reply tests/lib/test_schema.py]: done

File: tests/lib/test_schema.py (hunk 11 above)
Comment (perlpunk) [reply tests/lib/test_schema.py]: done, habit from perl ;-)

File: tests/lib/test_schema.py (hunk 5 above)
Comment (perlpunk) [reply tests/lib/test_schema.py:8]: I did as I was used from Perl. We don't have booleans there ;-)
I couldn't get it to work with True/False, though, maybe I had a typo

File: tests/lib/test_schema.py [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+        'str':   [str,   check_str],
+        'int':   [int,   check_int],
+        'float': [float, check_float],
+        'inf':   [float, check_float],
+        'nan':   [float, check_float],
+        'bool':  [bool,  check_bool],
+    }
+    with open(skip_filename, 'rb') as file:
+        skipdata = yaml.load(file, Loader=yaml.SafeLoader)
+    skip_load = skipdata['load'];
```
Comment (Thom1729) [inline tests/lib/test_schema.py:64]:
```suggestion
    skip_load = skipdata['load']
```

File: tests/lib/test_schema.py [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+    skip_dump = skipdata['dump'];
```
Comment (Thom1729) [inline tests/lib/test_schema.py:65]:
```suggestion
    skip_dump = skipdata['dump']
```

File: tests/lib/test_schema.py [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+    if verbose:
+        print(skip_load)
+    with open(data_filename, 'rb') as file:
+        tests = yaml.load(file, Loader=yaml.SafeLoader)
+
+    i = 0;
```
Comment (Thom1729) [inline tests/lib/test_schema.py:71]: Redundant with the `enumerate`.

```suggestion
```

File: tests/lib/test_schema.py [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+    fail = 0;
```
Comment (Thom1729) [inline tests/lib/test_schema.py:72]:
```suggestion
    fail = 0
```

File: tests/lib/test_schema.py [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+        else:
+            print("Compare: >>" + dump + "<< >>" + exp_dump + "<<");
+            fail+=1
+            _fail(input, test)
+
+#        if i >= 80:
+#            break
+
+    if fail > 0:
+        print("Failed " + str(fail) + " / " + str(i) + " tests");
```
Comment (Thom1729) [inline tests/lib/test_schema.py:140]:
```suggestion
        print("Failed " + str(fail) + " / " + str(i) + " tests")
```

File: tests/lib/test_schema.py [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+        assert(False)
+    else:
+        print("Passed " + str(i) + " tests");
```
Comment (Thom1729) [inline tests/lib/test_schema.py:143]:
```suggestion
        print("Passed " + str(i) + " tests")
```

File: tests/lib/test_schema.py [hunk 18]
```diff
@@ -0,0 +1,152 @@
...
+    print("Skipped " + str(len(skip_load)) + " load tests");
```
Comment (Thom1729) [inline tests/lib/test_schema.py:144]:
```suggestion
    print("Skipped " + str(len(skip_load)) + " load tests")
```

File: tests/lib/test_schema.py [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+    print("Skipped " + str(len(skip_dump)) + " dump tests");
```
Comment (Thom1729) [inline tests/lib/test_schema.py:145]:
```suggestion
    print("Skipped " + str(len(skip_dump)) + " dump tests")
```

**Review by Thom1729 — APPROVED** [review]

File: tests/lib/test_schema.py (hunk 12 above)
Comment (perlpunk) [reply tests/lib/test_schema.py]: done, all semicolons are removed =)

**Review by ingydotnet — COMMENTED** [review]
LGTM

**Pull request merged**